firewall/privacy_mapper.go

```diff
@@ -1087,6 +1094,80 @@ func handleClosedChannelsResponse(db firewalldb.PrivacyMapDB,
 	}
 }
 
+func handleBatchOpenChannelRequest(db firewalldb.PrivacyMapDB,
+	flags session.PrivacyFlags,
+	_ func(int) (int, error)) func(ctx context.Context,
+	r *lnrpc.BatchOpenChannelRequest) (proto.Message, error) {
+
+	return func(_ context.Context, r *lnrpc.BatchOpenChannelRequest) (
+		proto.Message, error) {
+
+		err := db.Update(func(tx firewalldb.PrivacyMapTx) error {
+			// Modify channels in place.
+			for _, c := range r.Channels {
+				var err error
+
+				// Note, this only works if the pubkey alias was
+				// already created via other calls, e.g. via
+				// ListChannels or GetInfo or the like.
+				if !flags.Contains(session.ClearPubkeys) {
+					c.NodePubkey, err = firewalldb.RevealBytes(
+						tx, c.NodePubkey,
+					)
+					if err != nil {
+						return err
+					}
+				}
+			}
+
+			return nil
+		})
```

Comment:
a thought popped up while looking at this commit: I wonder if we'd perhaps want to add a "Private chans only" option to the ChannelConstraint? for privacy reasons

Reply:
good idea! I added two fields to the rule, `PrivateAllowed` and `PublicAllowed` to be able to specify exactly which channel types are allowed, because one may want one or the other